app/middleware/getGps.js

```diff
@@ -0,0 +1,46 @@
+const MongoClient = require('mongodb').MongoClient;
+const log = require('../lib/logger');
+const constants = require('../lib/constants');
+const config = require('../../config/config').mongodb;
+const VError = require('verror').VError;
+
+function getGps(req, res, next) {
+  const connectionString = config.connectionString;
+
+  MongoClient.connect(connectionString).then((db) => {
+    log.info(`Connected to ${connectionString}`);
+
+    const collection = db.collection(config.collection);
+
+    const searchTerm = res.locals.search;
+    const limits = constants.numberOfNearbyResults;
+
+    collection.find({ name: new RegExp(searchTerm) }).limit(limits).toArray((errSearch, docs) => {
+      if (errSearch) {
+        const errMsg = 'MongoDB error while searching';
+        log.error({ err: new VError(errSearch, errMsg) }, errMsg);
+        next(errSearch);
+      }
+
+      log.debug(`Found ${docs.length} results for search term ${searchTerm}`);
+
+      // eslint-disable-next-line no-param-reassign
+      res.locals.gps = docs;
+
+      db.close((errClose, result) => {
```

Comment:
Unfortunately, after some head banging this has to be fixed outside of this PR https://github.com/nhsuk/gp-finder/issues/9 if this would be to progress